frontend/src/streamDetails/StreamStatisticsCard.jsx

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Grid } from "@material-ui/core";
+
+import DashboardCard from "../general/dashboard/DashboardCard";
+import SimpleTable from "../general/simpleTable/SimpleTable";
+import { getStreamStatistics } from "../api/StreamApi";
+import ButtonInfo from "../general/dashboard/ButtonInfo";
+
+export default class StreamStatisticsCard extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      stats: []
+    };
+    this.streamId = props.streamId;
+    this.handleStatsChange = this.handleStatsChange.bind(this);
+    this.getProperties = this.getProperties.bind(this);
+    this.getButton = this.getButton.bind(this);
+  }
+
+  componentDidMount() {
+    getStreamStatistics(this.streamId).then(this.handleStatsChange);
+  }
+
+  handleStatsChange(stats) {
+    this.setState({
+      stats
+    });
+  }
+
+  getProperties() {
+    const { stats } = this.state;
+    if (stats.length === 0) return {};
+    return {
+      Time: stats.time,
+      "Round-Trip-Time": stats.link.rtt,
+      "Packets Retransmitted": stats.send.packetsRetransmitted,
+      "Packets Dropped": stats.send.packetsDropped
```

Comment:
is stats an object or an array? the `.length` makes it seem like an array (as does the initial value set in the constructor) but here you're reading `stats.time`

Reply:
it's an object but i'm not sure how to go about setting the initial state correctly for no object
and then i'm not sure how to  check that stats is undefined in this piece of code. trying stuff like `=== undefined` `typeof stats === 'undefined'` and `=== null` all fail to load the page..;;;;

Reply:
you could make it an empty object `{}` and then i THINK there's an object.isEmpty but i'm not sure.
or, if you want to set it to `null `or `undefined`, then you can just use the value as a boolean and use it in an `if` statement or whatever since `null `and `undefined `convert to `false`

Reply:
this just in ! i'm stupid. that was the easiest thing to do in my life
1c54bb0